Build failure with clang-8.0.1

***PCSX2 version:***
https://github.com/PCSX2/pcsx2/commit/99f814d3764bc4ce87491ebeedac84762b3d68f3


***Description of the issue:***
When building pcsx2 with clang the build will fail.

```
[ 57%] Building CXX object plugins/GSdx/CMakeFiles/GSdx.dir/PSX/GPU.cpp.o
cd /tmp/SBo/pcsx2/build/plugins/GSdx && /usr/local/bin/clang++ -m32  -DDISABLE_BUILD_DATE -DDOC_DIR_COMPILATION=/usr/doc/pcsx2-2019.08.26_99f814d -DGAMEINDEX_DIR_COMPILATION=/usr/share/games/PCSX2 -DGSdx_EXPORTS -DPLUGIN_DIR_COMPILATION=/usr/lib/games/PCSX2 -DWXUSINGDLL -DXDG_STD -D_ARCH_32=1 -D_FILE_OFFSET_BITS=64 -D_M_X86=1 -D_M_X86_32=1 -D__WXGTK__ -I/usr/include/SDL2 -I/usr/include/freetype2 -I/usr/include/glib-2.0 -I/usr/lib32/glib-2.0/include -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/gtk-2.0 -I/usr/lib32/gtk-2.0/include -I/usr/include/libxml2 -I/usr/include/harfbuzz -I/tmp/SBo/pcsx2/common/include -I/tmp/SBo/pcsx2/common/include/Utilities -I/tmp/SBo/pcsx2/common/include/x86emitter -I/tmp/SBo/pcsx2/build/common/include -I/tmp/SBo/pcsx2/build/plugins/GSdx -I/tmp/SBo/pcsx2/plugins/GSdx/. -isystem /usr/lib/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0  -O2 -msse -msse2 -mfxsr -mxsave -march=i686 -pipe -fvisibility=hidden -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse -Wall -Wextra -Wno-attributes -Wno-unused-function -Wno-unused-parameter -Wno-missing-field-initializers -Wno-overloaded-virtual -Wno-unused-value   -Wno-deprecated-register -Wno-c++14-extensions -Wstrict-aliasing -Wstrict-overflow=1  -D_FORTIFY_SOURCE=2  -Wformat -Wformat-security -DNDEBUG  -O2    -std=c++11 -Wno-invalid-offsetof -pthread   -fno-operator-names -Wno-unknown-pragmas -Wno-parentheses -Wunused-variable  -fno-operator-names -Wno-unknown-pragmas -Wno-parentheses -Wunused-variable -DXDG_STD -o CMakeFiles/GSdx.dir/PSX/GPU.cpp.o -c /tmp/SBo/pcsx2/plugins/GSdx/PSX/GPU.cpp
In file included from /tmp/SBo/pcsx2/plugins/GSdx/PSX/GPU.cpp:23:
In file included from /tmp/SBo/pcsx2/plugins/GSdx/./GSdx.h:25:
In file included from /tmp/SBo/pcsx2/plugins/GSdx/./GS.h:40:
In file included from /tmp/SBo/pcsx2/plugins/GSdx/./GSVector.h:100:
/tmp/SBo/pcsx2/plugins/GSdx/./GSVector4i.h:769:21: error: argument value -4 is outside the valid range [0, 255] [-Wargument-outside-range]
                return GSVector4i(_mm_srli_si128(m, i));
                                  ^~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib64/clang/8.0.1/include/emmintrin.h:3057:12: note: expanded from macro '_mm_srli_si128'
  (__m128i)__builtin_ia32_psrldqi128_byteshift((__v2di)(__m128i)(a), (int)(imm))
           ^                                                         ~~~~~~~~~~
/tmp/SBo/pcsx2/plugins/GSdx/./GSVector4i.h:783:28: note: in instantiation of function template specialization 'GSVector4i::srl<-4>' requested here
                else if(i < 32) return v.srl<i - 16>();
                                         ^
/tmp/SBo/pcsx2/plugins/GSdx/./GSBlock.h:1402:12: note: in instantiation of function template specialization 'GSVector4i::srl<12>' requested here
                        v4 = v4.srl<12>(v5);
                                ^
In file included from /tmp/SBo/pcsx2/plugins/GSdx/PSX/GPU.cpp:23:
In file included from /tmp/SBo/pcsx2/plugins/GSdx/./GSdx.h:25:
In file included from /tmp/SBo/pcsx2/plugins/GSdx/./GS.h:40:
In file included from /tmp/SBo/pcsx2/plugins/GSdx/./GSVector.h:100:
/tmp/SBo/pcsx2/plugins/GSdx/./GSVector4i.h:769:21: error: argument value -8 is outside the valid range [0, 255] [-Wargument-outside-range]
                return GSVector4i(_mm_srli_si128(m, i));
                                  ^~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib64/clang/8.0.1/include/emmintrin.h:3057:12: note: expanded from macro '_mm_srli_si128'
  (__m128i)__builtin_ia32_psrldqi128_byteshift((__v2di)(__m128i)(a), (int)(imm))
           ^                                                         ~~~~~~~~~~
/tmp/SBo/pcsx2/plugins/GSdx/./GSVector4i.h:783:28: note: in instantiation of function template specialization 'GSVector4i::srl<-8>' requested here
                else if(i < 32) return v.srl<i - 16>();
                                         ^
/tmp/SBo/pcsx2/plugins/GSdx/./GSBlock.h:1404:12: note: in instantiation of function template specialization 'GSVector4i::srl<8>' requested here
                        v4 = v5.srl<8>(v6);
                                ^
2 errors generated.
make[2]: *** [plugins/GSdx/CMakeFiles/GSdx.dir/build.make:91: plugins/GSdx/CMakeFiles/GSdx.dir/PSX/GPU.cpp.o] Error 1
make[2]: Leaving directory '/tmp/SBo/pcsx2/build'
make[1]: *** [CMakeFiles/Makefile2:678: plugins/GSdx/CMakeFiles/GSdx.dir/all] Error 2
make[1]: Leaving directory '/tmp/SBo/pcsx2/build'
make: *** [Makefile:130: all] Error 2
```
Full build log: [pcsx2.txt](https://github.com/PCSX2/pcsx2/files/3548391/pcsx2.txt)


***How to reproduce the issue:***
```
export CC=clang
export CXX=clang++
# build pcsx2
```

***Last known version to work:***
Unknown, I have not tested pcsx2 with clang before.


***PC specifications:***
OS: `Slackware64-current`
clang: `8.0.1`

With `clang-9.0.0` it fails differently.
```
samu: job failed: /usr/local/bin/clang++ -m32  -DDISABLE_BUILD_DATE -DDOC_DIR_COMPILATION=/usr/doc/pcsx2-2019.09.29_45687cd -DGAMEINDEX_DIR_COMPILATION=/usr/share/games/PCSX2 -DPLUGIN_DIR_COMPILATION=/usr/lib/games/PCSX2 -DUSBnull_0_7_0_EXPORTS -DWXUSINGDLL -D_ARCH_32=1 -D_FILE_OFFSET_BITS=64 -D_M_X86=1 -D_M_X86_32=1 -D__WXGTK__ -I/usr/include/SDL2 -I/usr/include/freetype2 -I/usr/include/glib-2.0 -I/usr/lib32/glib-2.0/include -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/gtk-2.0 -I/usr/lib32/gtk-2.0/include -I/usr/include/libxml2 -I/usr/include/harfbuzz -I../common/include -I../common/include/Utilities -I../common/include/x86emitter -Icommon/include -I../plugins/USBnull/. -isystem /usr/lib/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0 -O2 -msse -msse2 -mfxsr -mxsave -march=i686 -pipe -fvisibility=hidden -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse -Wall -Wextra -Wno-attributes -Wno-unused-function -Wno-unused-parameter -Wno-missing-field-initializers -Wno-overloaded-virtual -Wno-unused-value   -Wno-deprecated-register -Wno-c++14-extensions -Wstrict-aliasing -Wstrict-overflow=1  -D_FORTIFY_SOURCE=2  -Wformat -Wformat-security -DNDEBUG  -O2    -std=c++11 -Wno-invalid-offsetof -pthread    -fvisibility=hidden -Wall -Wno-parentheses -MD -MT plugins/USBnull/CMakeFiles/USBnull-0.7.0.dir/USB.cpp.o -MF plugins/USBnull/CMakeFiles/USBnull-0.7.0.dir/USB.cpp.o.d -o plugins/USBnull/CMakeFiles/USBnull-0.7.0.dir/USB.cpp.o -c ../plugins/USBnull/USB.cpp
samu: subcommand failed
[1/789] Building CXX object plugins/USBnull/CMakeFiles/USBnull-0.7.0.dir/USB.cpp.o
In file included from ../plugins/USBnull/USB.cpp:20:
In file included from ../plugins/USBnull/./USB.h:22:
In file included from ../common/include/PS2Edefs.h:49:
In file included from ../common/include/Pcsx2Defs.h:36:
../common/include/x86emitter/x86_intrin.h:73:38: error: redefinition of '__rord' as different kind of symbol
[[maybe_unused]] static unsigned int _rotr(unsigned int x, int s)
                                     ^
/usr/bin/../lib64/clang/9.0.0/include/ia32intrin.h:357:20: note: expanded from macro '_rotr'
#define _rotr(a,b) __rord((a), (b))
                   ^
/usr/bin/../lib64/clang/9.0.0/include/ia32intrin.h:330:1: note: previous definition is here
__rord(unsigned int __X, int __C) {
^
In file included from ../plugins/USBnull/USB.cpp:20:
In file included from ../plugins/USBnull/./USB.h:22:
In file included from ../common/include/PS2Edefs.h:49:
In file included from ../common/include/Pcsx2Defs.h:36:
../common/include/x86emitter/x86_intrin.h:73:57: error: expected ')'
[[maybe_unused]] static unsigned int _rotr(unsigned int x, int s)
                                                        ^
../common/include/x86emitter/x86_intrin.h:73:38: note: to match this '('
[[maybe_unused]] static unsigned int _rotr(unsigned int x, int s)
                                     ^
/usr/bin/../lib64/clang/9.0.0/include/ia32intrin.h:357:27: note: expanded from macro '_rotr'
#define _rotr(a,b) __rord((a), (b))
                          ^
In file included from ../plugins/USBnull/USB.cpp:20:
In file included from ../plugins/USBnull/./USB.h:22:
In file included from ../common/include/PS2Edefs.h:49:
In file included from ../common/include/Pcsx2Defs.h:36:
../common/include/x86emitter/x86_intrin.h:73:38: error: expected expression
[[maybe_unused]] static unsigned int _rotr(unsigned int x, int s)
                                     ^
/usr/bin/../lib64/clang/9.0.0/include/ia32intrin.h:357:30: note: expanded from macro '_rotr'
#define _rotr(a,b) __rord((a), (b))
                             ^
In file included from ../plugins/USBnull/USB.cpp:20:
In file included from ../plugins/USBnull/./USB.h:22:
In file included from ../common/include/PS2Edefs.h:49:
In file included from ../common/include/Pcsx2Defs.h:36:
../common/include/x86emitter/x86_intrin.h:73:64: error: expected ')'
[[maybe_unused]] static unsigned int _rotr(unsigned int x, int s)
                                                               ^
../common/include/x86emitter/x86_intrin.h:73:38: note: to match this '('
[[maybe_unused]] static unsigned int _rotr(unsigned int x, int s)
                                     ^
/usr/bin/../lib64/clang/9.0.0/include/ia32intrin.h:357:32: note: expanded from macro '_rotr'
#define _rotr(a,b) __rord((a), (b))
                               ^
In file included from ../plugins/USBnull/USB.cpp:20:
In file included from ../plugins/USBnull/./USB.h:22:
In file included from ../common/include/PS2Edefs.h:49:
In file included from ../common/include/Pcsx2Defs.h:36:
../common/include/x86emitter/x86_intrin.h:73:38: error: expected expression
[[maybe_unused]] static unsigned int _rotr(unsigned int x, int s)
                                     ^
/usr/bin/../lib64/clang/9.0.0/include/ia32intrin.h:357:35: note: expanded from macro '_rotr'
#define _rotr(a,b) __rord((a), (b))
                                  ^
In file included from ../plugins/USBnull/USB.cpp:20:
In file included from ../plugins/USBnull/./USB.h:22:
In file included from ../common/include/PS2Edefs.h:49:
In file included from ../common/include/Pcsx2Defs.h:36:
../common/include/x86emitter/x86_intrin.h:73:66: error: expected ';' after top level declarator
[[maybe_unused]] static unsigned int _rotr(unsigned int x, int s)
                                                                 ^
                                                                 ;
6 errors generated.
```

With the above referenced PR I am back to where I am started with `clang-9.0.0`.
```
[189/717] Building CXX object plugins/GSdx/CMakeFiles/GSdx.dir/Window/GSWnd.cpp.o
[190/717] Building CXX object plugins/GSdx/CMakeFiles/GSdx.dir/Window/GSSetting.cpp.o
[191/717] Building CXX object plugins/GSdx/CMakeFiles/GSdx.dir/Renderers/OpenCL/GSRendererCL.cpp.o
In file included from ../plugins/GSdx/Renderers/OpenCL/GSRendererCL.cpp:23:
In file included from ../plugins/GSdx/Renderers/OpenCL/GSRendererCL.h:24:
In file included from ../plugins/GSdx/./Renderers/Common/GSRenderer.h:24:
In file included from ../plugins/GSdx/./GSdx.h:25:
In file included from ../plugins/GSdx/./GS.h:40:
In file included from ../plugins/GSdx/./GSVector.h:100:
../plugins/GSdx/./GSVector4i.h:769:21: error: argument value -4 is outside the valid range [0, 255] [-Wargument-outside-range]
                return GSVector4i(_mm_srli_si128(m, i));
                                  ^~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib64/clang/9.0.0/include/emmintrin.h:3038:12: note: expanded from macro '_mm_srli_si128'
  (__m128i)__builtin_ia32_psrldqi128_byteshift((__v2di)(__m128i)(a), (int)(imm))
           ^                samu: subcommand failed
                                         ~~~~~~~~~~
../plugins/GSdx/./GSVector4i.h:783:28: note: in instantiation of function template specialization 'GSVector4i::srl<-4>' requested here
                else if(i < 32) return v.srl<i - 16>();
                                         ^
../plugins/GSdx/./GSBlock.h:1402:12: note: in instantiation of function template specialization 'GSVector4i::srl<12>' requested here
                        v4 = v4.srl<12>(v5);
                                ^
In file included from ../plugins/GSdx/Renderers/OpenCL/GSRendererCL.cpp:23:
In file included from ../plugins/GSdx/Renderers/OpenCL/GSRendererCL.h:24:
In file included from ../plugins/GSdx/./Renderers/Common/GSRenderer.h:24:
In file included from ../plugins/GSdx/./GSdx.h:25:
In file included from ../plugins/GSdx/./GS.h:40:
In file included from ../plugins/GSdx/./GSVector.h:100:
../plugins/GSdx/./GSVector4i.h:769:21: error: argument value -8 is outside the valid range [0, 255] [-Wargument-outside-range]
                return GSVector4i(_mm_srli_si128(m, i));
                                  ^~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib64/clang/9.0.0/include/emmintrin.h:3038:12: note: expanded from macro '_mm_srli_si128'
  (__m128i)__builtin_ia32_psrldqi128_byteshift((__v2di)(__m128i)(a), (int)(imm))
           ^                                                         ~~~~~~~~~~
../plugins/GSdx/./GSVector4i.h:783:28: note: in instantiation of function template specialization 'GSVector4i::srl<-8>' requested here
                else if(i < 32) return v.srl<i - 16>();
                                         ^
../plugins/GSdx/./GSBlock.h:1404:12: note: in instantiation of function template specialization 'GSVector4i::srl<8>' requested here
                        v4 = v5.srl<8>(v6);
                                ^
2 errors generated.
```

--- a/.github/workflows/linux-workflow.yml
+++ b/.github/workflows/linux-workflow.yml
@@ -34,12 +34,27 @@ jobs:
       # Prevent one build from failing everything (although maybe those should be included as experimental builds instead)
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04]
-        platform: [x86, x64]
-        compiler: [gcc, clang]
-        experimental: [false]
+        include:
+          - os: ubuntu-20.04
+            platform: x86
+            compiler: gcc
+            experimental: false
+          - os: ubuntu-20.04
+            platform: x64
+            compiler: gcc
+            experimental: false
+          - os: ubuntu-20.04
+            platform: x86
+            compiler: clang
+            experimental: false
+          - os: ubuntu-20.04
+            platform: x86
+            compiler: gcc
+            cmakeflags: -DCMAKE_DISABLE_PRECOMPILE_HEADERS=ON
+            detail: -nopch
+            experimental: true
 
-    name: ${{ matrix.os }}-${{ matrix.platform }}-${{ matrix.compiler }}
+    name: ${{ matrix.os }}-${{ matrix.platform }}-${{ matrix.compiler }}${{ matrix.detail }}
     runs-on: ${{ matrix.os }}
     continue-on-error: ${{ matrix.experimental }}
     # Set some sort of timeout in the event of run-away builds.  We are limited on concurrent jobs so, get rid of them.
@@ -84,9 +99,9 @@ jobs:
         uses: actions/cache@v2
         with:
           path: .ccache
-          key: ${{ matrix.os }}-${{ matrix.platform }}-${{ matrix.compiler }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+          key: ${{ matrix.os }}-${{ matrix.platform }}-${{ matrix.compiler }}${{ matrix.detail }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
           restore-keys: |
-            ${{ matrix.os }}-${{ matrix.platform }}-${{ matrix.compiler }}-ccache-
+            ${{ matrix.os }}-${{ matrix.platform }}-${{ matrix.compiler }}${{ matrix.detail }}-ccache-
 
       - name: Install Packages
         env:
@@ -98,6 +113,7 @@ jobs:
         env:
           PLATFORM: ${{ matrix.platform }}
           COMPILER: ${{ matrix.compiler }}
+          ADDITIONAL_CMAKE_ARGS: ${{ matrix.cmakeflags }}
         run: ./.github/workflows/scripts/linux/generate-cmake.sh
 
       - name: Build PCSX2

--- a/.github/workflows/scripts/linux/generate-cmake.sh
+++ b/.github/workflows/scripts/linux/generate-cmake.sh
@@ -17,6 +17,7 @@ echo "Additional CMake Args - ${ADDITIONAL_CMAKE_ARGS}"
 
 # Generate CMake into ./build
 cmake \
+-D CMAKE_CXX_COMPILER_LAUNCHER=ccache \
 -D CMAKE_BUILD_TYPE=Release \
 -D BUILD_REPLAY_LOADERS=TRUE \
 -D CMAKE_BUILD_PO=FALSE \

--- a/.github/workflows/scripts/linux/generate-cmake.sh
+++ b/.github/workflows/scripts/linux/generate-cmake.sh
@@ -18,6 +18,7 @@ echo "Additional CMake Args - ${ADDITIONAL_CMAKE_ARGS}"
 # Generate CMake into ./build
 cmake \
 -D CMAKE_CXX_COMPILER_LAUNCHER=ccache \
+-D DISABLE_ADVANCE_SIMD=TRUE \
 -D CMAKE_BUILD_TYPE=Release \
 -D BUILD_REPLAY_LOADERS=TRUE \
 -D CMAKE_BUILD_PO=FALSE \

--- a/.github/workflows/scripts/linux/generate-cmake.sh
+++ b/.github/workflows/scripts/linux/generate-cmake.sh
@@ -9,9 +9,9 @@ else
   export CC=clang
   export CXX=clang++
 fi
-ADDITIONAL_CMAKE_ARGS=""
+
 if [ "${PLATFORM}" = "x86" ]; then
-  ADDITIONAL_CMAKE_ARGS="-D CMAKE_TOOLCHAIN_FILE=cmake/linux-compiler-i386-multilib.cmake"
+  ADDITIONAL_CMAKE_ARGS="$ADDITIONAL_CMAKE_ARGS -DCMAKE_TOOLCHAIN_FILE=cmake/linux-compiler-i386-multilib.cmake"
 fi
 echo "Additional CMake Args - ${ADDITIONAL_CMAKE_ARGS}"
 

--- a/common/include/Utilities/SafeArray.h
+++ b/common/include/Utilities/SafeArray.h
@@ -15,6 +15,8 @@
 
 #pragma once
 
+#include "Utilities/Dependencies.h"
+
 // pxUSE_SECURE_MALLOC - enables bounds checking on scoped malloc allocations.
 
 #ifndef pxUSE_SECURE_MALLOC

--- a/common/include/Utilities/pxEvents.h
+++ b/common/include/Utilities/pxEvents.h
@@ -15,6 +15,8 @@
 
 #pragma once
 
+#include <wx/event.h>
+
 wxDECLARE_EVENT(pxEvt_StartIdleEventTimer, wxCommandEvent);
 wxDECLARE_EVENT(pxEvt_DeleteObject, wxCommandEvent);
 wxDECLARE_EVENT(pxEvt_DeleteThread, wxCommandEvent);

--- a/common/include/x86emitter/tools.h
+++ b/common/include/x86emitter/tools.h
@@ -15,6 +15,8 @@
 
 #pragma once
 
+#include "Utilities/Dependencies.h"
+
 enum x86VendorType {
     x86Vendor_Intel = 0,
     x86Vendor_AMD = 1,

--- a/pcsx2/Config.h
+++ b/pcsx2/Config.h
@@ -17,6 +17,10 @@
 
 #include "x86emitter/tools.h"
 
+#include "Utilities/FixedPointTypes.h"
+#include "Utilities/General.h"
+#include <wx/filename.h>
+
 class IniInterface;
 
 enum PluginsEnum_t

--- a/pcsx2/PathDefs.h
+++ b/pcsx2/PathDefs.h
@@ -15,6 +15,8 @@
 
 #pragma once
 
+#include "Utilities/Path.h"
+
 enum FoldersEnum_t
 {
 	// FIXME : Plugins and Settings folders are no longer part of the user-local

--- a/pcsx2/Plugins.h
+++ b/pcsx2/Plugins.h
@@ -18,9 +18,11 @@
 #define PLUGINtypedefs
 #define PLUGINfuncs
 
+#include "Config.h"
 #include "PS2Edefs.h"
 #include "PluginCallbacks.h"
 
+#include <Utilities/MemcpyFast.h>
 #include "Utilities/Threading.h"
 
 #include <wx/dynlib.h>

--- a/pcsx2/USB/configuration.h
+++ b/pcsx2/USB/configuration.h
@@ -15,11 +15,12 @@
 
 #pragma once
 
+#include "platcompat.h"
+#include <wx/string.h>
 #include <vector>
 #include <string>
 #include <map>
 #include <sstream>
-#include "platcompat.h"
 
 #define RESULT_CANCELED 0
 #define RESULT_OK 1

--- a/pcsx2/USB/linux/config.cpp
+++ b/pcsx2/USB/linux/config.cpp
@@ -13,12 +13,13 @@
  *  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "config.h"
+
 #include "../configuration.h"
 #include "../deviceproxy.h"
 #include "../usb-pad/padproxy.h"
 #include "../usb-mic/audiodeviceproxy.h"
-
-#include "config.h"
+#include "Utilities/Console.h"
 
 void SysMessage_stderr(const char* fmt, ...)
 {

--- a/pcsx2/USB/usb-eyetoy/cam-linux.cpp
+++ b/pcsx2/USB/usb-eyetoy/cam-linux.cpp
@@ -13,6 +13,13 @@
  *  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "cam-linux.h"
+#include "usb-eyetoy-webcam.h"
+#include "jpgd/jpgd.h"
+#include "jo_mpeg.h"
+#include "../gtk.h"
+#include "Utilities/Console.h"
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -27,13 +34,6 @@
 #include <unistd.h>
 
 #include <linux/videodev2.h>
-
-#include "../gtk.h"
-
-#include "cam-linux.h"
-#include "usb-eyetoy-webcam.h"
-#include "jpgd/jpgd.h"
-#include "jo_mpeg.h"
 
 GtkWidget* new_combobox(const char* label, GtkWidget* vbox); // src/linux/config-gtk.cpp
 

--- a/pcsx2/USB/usb-hid/evdev/evdev.h
+++ b/pcsx2/USB/usb-hid/evdev/evdev.h
@@ -14,13 +14,14 @@
  */
 
 #pragma once
+#include "../usb-hid.h"
 #include "../../linux/util.h"
+#include "Utilities/Console.h"
 #include <linux/input.h>
 #include <unistd.h>
 #include <dirent.h>
 #include <thread>
 #include <atomic>
-#include "../usb-hid.h"
 
 namespace usb_hid
 {

--- a/pcsx2/USB/usb-msd/usb-msd-gtk.cpp
+++ b/pcsx2/USB/usb-msd/usb-msd-gtk.cpp
@@ -17,6 +17,7 @@
 #include "../linux/ini.h"
 #include "../configuration.h"
 #include "../gtk.h"
+#include "Utilities/Console.h"
 
 namespace usb_msd
 {

--- a/pcsx2/USB/usb-pad/evdev/evdev.h
+++ b/pcsx2/USB/usb-pad/evdev/evdev.h
@@ -14,14 +14,15 @@
  */
 
 #pragma once
+#include "evdev-ff.h"
+#include "shared.h"
 #include "../../linux/util.h"
+#include "../../readerwriterqueue/readerwriterqueue.h"
+#include "Utilities/Console.h"
 //#include <dirent.h> //gtk.h pulls in?
 #include <thread>
 #include <array>
 #include <atomic>
-#include "evdev-ff.h"
-#include "shared.h"
-#include "../../readerwriterqueue/readerwriterqueue.h"
 
 namespace usb_pad
 {

--- a/pcsx2/USB/usb-pad/joydev/joydev.cpp
+++ b/pcsx2/USB/usb-pad/joydev/joydev.cpp
@@ -14,9 +14,10 @@
  */
 
 #include "joydev.h"
+#include "../../linux/util.h"
+#include "Utilities/Console.h"
 #include <cassert>
 #include <sstream>
-#include "../../linux/util.h"
 
 namespace usb_pad
 {

--- a/pcsx2/USB/usb-pad/joydev/joydev.h
+++ b/pcsx2/USB/usb-pad/joydev/joydev.h
@@ -17,6 +17,7 @@
 #include "../../linux/util.h"
 #include "../evdev/evdev-ff.h"
 #include "../evdev/shared.h"
+#include "Utilities/Console.h"
 
 namespace usb_pad
 {

--- a/pcsx2/gui/AppConfig.h
+++ b/pcsx2/gui/AppConfig.h
@@ -16,8 +16,14 @@
 #pragma once
 
 #include "AppForwardDefs.h"
+#include "Config.h"
 #include "PathDefs.h"
 #include "CDVD/CDVDaccess.h"
+#include "Utilities/General.h"
+#include "Utilities/Path.h"
+
+#include <wx/colour.h>
+#include <wx/gdicmn.h>
 #include <memory>
 
 enum DocsModeType

--- a/pcsx2/gui/pxEventThread.h
+++ b/pcsx2/gui/pxEventThread.h
@@ -17,6 +17,8 @@
 
 #include "Utilities/PersistentThread.h"
 #include "Utilities/pxEvents.h"
+
+#include <wx/timer.h>
 #include <memory>
 
 // TODO!!  Make the system defined in this header system a bit more generic, and then move

--- a/plugins/GSdx/GSVector4i.h
+++ b/plugins/GSdx/GSVector4i.h
@@ -777,10 +777,11 @@ public:
 
 		#else
 
+		// The `& 0xF` keeps the compiler happy on cases that won't actually be hit
 		if(i == 0) return *this;
-		else if(i < 16) return srl<i>() | v.sll<16 - i>();
+		else if(i < 16) return srl<i & 0xF>() | v.sll<(16 - i) & 0xF>();
 		else if(i == 16) return v;
-		else if(i < 32) return v.srl<i - 16>();
+		else if(i < 32) return v.srl<(i - 16) & 0xF>();
 		else return zero();
 
 		#endif

